test/CMakeLists.txt

```diff
@@ -7,72 +7,163 @@ FetchContent_Declare(doctest
 )
 FetchContent_MakeAvailable(doctest)
 
-add_library(sfml-test-main STATIC
-    DoctestMain.cpp
-    TestUtilities/SystemUtil.hpp
-    TestUtilities/SystemUtil.cpp
-    TestUtilities/WindowUtil.hpp
-    TestUtilities/WindowUtil.cpp
-    TestUtilities/GraphicsUtil.hpp
-    TestUtilities/GraphicsUtil.cpp
-)
-target_include_directories(sfml-test-main PUBLIC TestUtilities)
-target_compile_definitions(sfml-test-main PUBLIC DOCTEST_CONFIG_REQUIRE_STRINGIFICATION_FOR_ALL_USED_TYPES)
-target_link_libraries(sfml-test-main PUBLIC SFML::System doctest::doctest)
-set_target_warnings(sfml-test-main)
-
-SET(SYSTEM_SRC
-    System/Angle.cpp
-    System/Clock.cpp
-    System/Err.cpp
-    System/FileInputStream.cpp
-    System/MemoryInputStream.cpp
-    System/Time.cpp
-    System/Vector2.cpp
-    System/Vector3.cpp
-)
-sfml_add_test(test-sfml-system "${SYSTEM_SRC}" SFML::System)
+# add a new static library which provides a main function and testing utilities for SFML types
+# example: sfml_add_test_utils(sfml-window-test-utils
+#                              SOURCES TestUtilities/WindowUtil.hpp TestUtilities/WindowUtil.cpp
+#                              DEPENDS SFML::Window sfml-system-test-utils)
```

Comment:
I'm not sure we need these comments anymore since all the users of this function reside in the same file. Same goes for `sfml_add_test`'s description comment.

Reply:
I don't see a big difference if the code using those functions is in the same file or not.
A small description doesn't hurt IMHO.